bob-access/src/authenticator/users_storage/config.rs

```diff
@@ -1,20 +1,40 @@
 use std::collections::HashMap;
+use std::fs::File;
 
 use crate::error::Error;
 
 use super::{Perms, User};
 
+use hex::FromHex;
+
 #[derive(Debug, Clone, Deserialize)]
-pub(super) struct ConfigUser {
+pub struct ConfigUser {
     pub(super) username: String,
-    pub(super) password: String,
+    pub(super) password: Option<String>,
+    pub(super) password_hash: Option<String>,
     pub(super) role: String,
 }
 
 #[derive(Debug, Deserialize)]
-pub(super) struct ConfigUsers {
+pub struct ConfigUsers {
     pub(super) roles: HashMap<String, Perms>,
     pub(super) users: Vec<ConfigUser>,
+    #[serde(default = "ConfigUsers::default_password_salt")]
+    pub password_salt: String,
```

Comment:
`pub(super)`?

Reply:
pub(super) limits field to be visible only inside /user_storage, so in case of password_salt it needs to be 'pub' (or can I make the field private and create getter method?)

Reply:
I just noticed, that you started using `ConfigUsers` outside of the current module. This is a bad design and should be reverted. You can store `password_salt` inside `UsersMap` struct and add getter to that struct